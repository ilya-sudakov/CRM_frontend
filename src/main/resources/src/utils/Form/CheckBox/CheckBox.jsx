import React from 'react';
import './CheckBox.scss';

const CheckBox = (props) => {
    return (
        <div className="checkbox">
            <label className="checkbox__container">
                {props.text}
                <input
                    type="checkbox"
                    name={props.name}
                    id={props.id}
                    value={props.value}
                    disabled={props.disabled}
                    onChange={(event) => {
                        const name = event.target.name;
                        const value = event.target.checked;
                        const id = event.target.id
                        props.onChange(value, name, id)
                    }}
                    checked={props.checked && props.checked}
                    defaultChecked={(props.defaultChecked !== undefined) ? props.defaultChecked : true}
                />
                <div class="checkbox__checkmark"></div>
            </label>
        </div>
    );
};

export default CheckBox;