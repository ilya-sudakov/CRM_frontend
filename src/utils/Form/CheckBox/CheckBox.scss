@import "../../../variables.scss";

.checkbox {
  width: 100%;
  margin-bottom: 15px;
  --checkbox__checkmark--size: 22px;

  /* Customize the label (the checkbox__checkbox-container) */
  .checkbox__container {
    display: flex;
    align-items: center;
    position: relative;
    width: 100%;
    cursor: pointer;
    font-size: 0.9rem;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    transition: background-color 100ms ease-in-out;
  }

  /* Hide the browser's default checkbox */
  .checkbox__container input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
    height: 0;
    width: 0;
  }

  /* Create a custom checkbox */
  .checkbox__checkmark {
    display: flex;
    top: 0;
    left: 0;
    height: var(--checkbox__checkmark--size);
    min-width: var(--checkbox__checkmark--size);
    border-radius: 3px;
    transition: background-color 100ms ease-in-out;
    margin-right: 5px;
    box-sizing: border-box;
    position: relative;

    border: 1px solid #cccccc;
  }

  /* On mouse-over, add a grey background color */
  .checkbox__container:hover input ~ .checkbox__checkmark {
    background-color: #f4f4f4;
  }

  /* When the checkbox is checked, add a blue background */
  .checkbox__container input:checked ~ .checkbox__checkmark {
    // background-color: #77c2ff;
    border-color: transparent;
    background-color: $main-color;
  }

  /* Create the checkmark/indicator (hidden when not checked) */
  .checkbox__checkmark:after {
    content: "";
    position: absolute;
    display: none;
  }

  /* Show the checkmark when checked */
  .checkbox__container input:checked ~ .checkbox__checkmark:after {
    display: block;
  }

  /* Style the checkmark/indicator */
  .checkbox__container .checkbox__checkmark:after {
    left: 7px;
    top: 2px;
    width: 5px;
    height: 10px;
    border: solid white;
    border-width: 0 2px 2px 0;
    -webkit-transform: rotate(45deg);
    -ms-transform: rotate(45deg);
    transform: rotate(45deg);
  }
  .checkbox__text {
    width: 100%;
    color: #555555;
  }
}
