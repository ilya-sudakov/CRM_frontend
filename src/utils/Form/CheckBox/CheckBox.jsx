import PropTypes from 'prop-types';
import { useEffect } from 'react';
import './CheckBox.scss';

const CheckBox = ({
  name = '',
  text = '',
  id,
  value,
  disabled = false,
  onChange,
  checked = false,
}) => {
  const handleChange = ({ target }) => {
    const name = target.name;
    const value = target.checked;
    const id = target.id;
    onChange(value, name, id);
  };

  return (
    <div className="checkbox">
      <label className="checkbox__container">
        <input
          type="checkbox"
          name={name}
          id={id}
          value={value}
          disabled={disabled}
          onChange={handleChange}
          checked={checked}
        />
        <div className="checkbox__checkmark"></div>
        <span className="checkbox__text">{text}</span>
      </label>
    </div>
  );
};

export default CheckBox;

CheckBox.propTypes = {
  name: PropTypes.string,
  text: PropTypes.string,
  id: PropTypes.number,
  value: PropTypes.bool,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
  checked: PropTypes.bool,
};
