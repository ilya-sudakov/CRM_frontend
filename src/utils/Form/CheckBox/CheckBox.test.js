import React from "react";
import CheckBox from "./CheckBox.jsx";
import { render, cleanup, fireEvent, screen } from "@testing-library/react";

describe("CheckBox component", () => {
  afterEach(cleanup);

  it("matches snapshot", () => {
    const { asFragment } = render(<CheckBox />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders w/o props", () => {
    render(<CheckBox />);
  });

  it("becomes checked", () => {
    let checked = false;
    render(
      <CheckBox
        name="test"
        checked={checked}
        onChange={(value) => (checked = value)}
      />
    );
    const checkbox = screen.getByRole("checkbox");
    fireEvent.click(checkbox);
    expect(checked).toBe(true);
  });
});
