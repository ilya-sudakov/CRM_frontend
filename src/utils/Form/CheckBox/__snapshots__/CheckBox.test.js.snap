// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="checkbox"
  >
    <label
      class="checkbox__container"
    >
      <input
        name=""
        type="checkbox"
        value=""
      />
      <div
        class="checkbox__checkmark"
      />
      <span
        class="checkbox__text"
      />
    </label>
  </div>
</DocumentFragment>
`;
